conformance/tests/gateway-with-attached-routes.go

```diff
@@ -76,5 +76,39 @@ var GatewayWithAttachedRoutes = suite.ConformanceTest{
 
 			kubernetes.GatewayStatusMustHaveListeners(t, s.Client, s.TimeoutConfig, gwNN, listeners)
 		})
+
+		t.Run("Gateway listener should not have route attached if route doesn't exist", func(t *testing.T) {
+			gwNN := types.NamespacedName{Name: "gateway-with-two-listeners-and-one-attached-route", Namespace: "gateway-conformance-infra"}
+			listeners := []v1beta1.ListenerStatus{
+				{
+					Name: v1beta1.SectionName("udp"),
+					SupportedKinds: []v1beta1.RouteGroupKind{{
+						Group: (*v1beta1.Group)(&v1beta1.GroupVersion.Group),
+						Kind:  v1beta1.Kind("UDPRoute"),
+					}},
+					Conditions: []metav1.Condition{{
+						Type:   string(v1beta1.ListenerConditionAccepted),
+						Status: metav1.ConditionTrue,
```

Comment:
contour does not support UDPRoute or the UDP protocol on a Listener so this Listener will never be accepted, could we switch to a different protocol?

Reply:
I don't mind switching it to TCP
@shaneutt @mlavacca, are you guys ok with that?

Reply:
Actually, I don't think we can be in a situation where `SupportGateway` as a feature implicit requires TCP or UDP awareness :thinking:

Reply:
> Actually, I don't think we can be in a situation where SupportGateway as a feature implicit requires TCP or UDP awareness 🤔

I agree with you, I see two ways of getting over this problem:
1. move this test into a new file, add TLS as a feature, and use TLS as the protocol for the listener (I'm suggesting TLS as we already have some TLS tests).
2. change this test to check that the `SectionName` use is respected (one gateway with two listeners and one HTTPRoute that references only one of them by means the `SectionName` field, should result in only one listener with `AttachedRoutes` incremented)

WDYT?

Reply:
I can do both :)
Let me know what you prefer

Reply:
@mlavacca I've updated the test according to your 2nd suggestion, if you want the first one too please do let me know.

Reply:
@shaneutt I'm curious to hear your thoughts on this one 🤔

Reply:
I like the sectionname approach since it is a core field 👍🏽

Reply:
seems like the test name will need to be updated since the protocol/route type isnt the differentiator now

Reply:
@sunjayBhatia I definitely agree. Changed the test name.
I'm resolving this conversion, but please do let me know if you think the new name is not descriptive enough.